persistence/gov.go

```diff
@@ -9,893 +14,89 @@ import (
+	}
+	_, err = conn.Exec(ctx, schema.InsertParams(genesis.DefaultParams(), p.Height))
+	return err
 }
 
-func (p PostgresContext) GetMessageUnpauseValidatorFeeOwner() ([]byte, error) {
-	return p.GetBytesParam(types.MessageUnpauseValidatorFeeOwner)
+func (p PostgresContext) GetIntParam(paramName string, height int64) (int, error) {
+	return GetParam[int](p, paramName, height)
 }
 
-func (p PostgresContext) GetMessageStakeServiceNodeFeeOwner() ([]byte, error) {
-	return p.GetBytesParam(types.MessageStakeServiceNodeFeeOwner)
+func (p PostgresContext) GetStringParam(paramName string, height int64) (string, error) {
+	return GetParam[string](p, paramName, height)
 }
 
-func (p PostgresContext) GetMessageEditStakeServiceNodeFeeOwner() ([]byte, error) {
-	return p.GetBytesParam(types.MessageEditStakeServiceNodeFeeOwner)
+func (p PostgresContext) GetBytesParam(paramName string, height int64) (param []byte, err error) {
+	return GetParam[[]byte](p, paramName, height)
 }
 
-func (p PostgresContext) GetMessageUnstakeServiceNodeFeeOwner() ([]byte, error) {
-	return p.GetBytesParam(types.MessageUnstakeServiceNodeFeeOwner)
+func (p PostgresContext) SetParam(paramName string, value interface{}) error {
+	switch t := value.(type) {
+	case int:
+		return SetParam(p, paramName, t)
+	case int32:
+		return SetParam(p, paramName, t)
+	case int64:
+		return SetParam(p, paramName, t)
+	case []byte:
+		return SetParam(p, paramName, t)
+	case string:
+		return SetParam(p, paramName, t)
+	default:
+		log.Fatalf("unhandled paramType %T for value %v", t, value)
+	}
+	return fmt.Errorf("how did you get here?")
 }
 
-func (p PostgresContext) GetMessagePauseServiceNodeFeeOwner() ([]byte, error) {
-	return p.GetBytesParam(types.MessagePauseServiceNodeFeeOwner)
+func (p PostgresContext) InitFlags() error {
+	//TODO: not implemented
+	return nil
 }
 
-func (p PostgresContext) GetMessageUnpauseServiceNodeFeeOwner() ([]byte, error) {
-	return p.GetBytesParam(types.MessageUnpauseServiceNodeFeeOwner)
+func (p PostgresContext) GetIntFlag(paramName string, height int64) (value int, enabled bool, err error) {
+	return GetFlag[int](p, paramName, height)
 }
 
-func (p PostgresContext) GetMessageChangeParameterFeeOwner() ([]byte, error) {
-	return p.GetBytesParam(types.MessageChangeParameterFeeOwner)
+func (p PostgresContext) GetStringFlag(paramName string, height int64) (value string, enabled bool, err error) {
+	return GetFlag[string](p, paramName, height)
 }
 
-func (p PostgresContext) GetServiceNodesPerSessionAt(height int64) (int, error) {
-	return p.GetIntParam(types.ServiceNodesPerSessionParamName)
+func (p PostgresContext) GetBytesFlag(paramName string, height int64) (value []byte, enabled bool, err error) {
+	return GetFlag[[]byte](p, paramName, height)
 }
 
-func (p PostgresContext) InitParams() error {
-	ctx, conn, err := p.GetCtxAndConnection()
-	if err != nil {
-		return err
+func (p PostgresContext) SetFlag(paramName string, value interface{}, enabled bool) error {
```

Comment:
NIT: Golang 1.18 introduce `any` so that may actually be more appropriate than `interface{}` here based on the answer in the comment here: https://stackoverflow.com/questions/72283932/go-difference-between-any-and-interface-in-varargs

Reply:
I agree, I like it more as `any` here too.

I would suggest raising a separate PR in which we replace `interface{}` with `any` across the board for consistency.

Ultimately:
```go
// any is an alias for interface{} and is equivalent to interface{} in all ways.
type any = interface{}
```
but perhaps it's more idiomatic especially for newcomers to Go

Reply:
Done: https://github.com/pokt-network/pocket/issues/156